### PR TITLE
Feature/2486/add search by user name on the chat page

### LINKED ISSUE #2486: (SP: 1) Fix search by user name on the Chat Page
## Description
A User should have the opportunity to find an appropriate user by name on the chat page. The input search text has to be maximum 30 characters long containing alphabetic characters (case insensitive), hyphen and a space is allowed. English and Ukrainian language are accepted. 

**Reproducible:** Always 

**Preconditions:** The user is logged in as tutor/student

**Steps to reproduce:**
1. Go to Chat Page
2. Type first letters of the user name in search field

**Expected result:** The user whose name matches with input letters is found

**Actual result:** No users found

Screenshot 2024-09-10 at 13 11 13

### PR DESCRIPTION
**Pull request by OliaHrytsak**
Now user can use search input to filter chats by member name!
https://github.com/user-attachments/assets/c7cf4ed2-ad86-4a3b-aae6-5c1669447118

Refactored failed tests in BookmarkedOffers.spec.jsx and CoursesFiltersDrawer.spec.jsx

### CODE REVIEW

**Comment by BohdanMylyi** [comment]
hi! fix sonar issue, please
image

**Comment by OliaHrytsak** [comment]
> hi! fix sonar issue, please image

Actually, this warning ia about spec.constants file, which sonar detected us redundant, but this mocks are used in several tests files and because of that it is impossible to remove them

image

I think, then additionall tests for chat will be added, this warning will disappear!

**Review by YaroslavChuiko — APPROVED** [review]

File: src/containers/chat/list-of-users-with-search/ListOfUsersWithSearch.tsx [hunk 1]
```diff
@@ -31,6 +31,10 @@ const ListOfUsersWithSearch: FC<ListOfUsersWithSearchProps> = ({
 
   const { userId } = useAppSelector((state) => state.appMain)
   const { t } = useTranslation()
+  console.log(userId)
```
Comment (Mav-Ivan) [inline src/containers/chat/list-of-users-with-search/ListOfUsersWithSearch.tsx]: Do we need this log ?

Comment (OliaHrytsak) [reply src/containers/chat/list-of-users-with-search/ListOfUsersWithSearch.tsx]: removed

**Review by Mav-Ivan — APPROVED** [review]

**Review by PavloDolia — APPROVED** [review]

**Comment by PavloDolia** [comment]
> > hi! fix sonar issue, please image
> 
> Actually, this warning ia about spec.constants file, which sonar detected us redundant, but this mocks are used in several tests files and because of that it is impossible to remove them
> 
> image
> 
> I think, then additionall tests for chat will be added, this warning will disappear!

You can remove 'spec' from the file name to fix this issue
(`tests/unit/pages/chat/ChatsMock.spec.constants.js` => `tests/unit/pages/chat/ChatsMock.constants.js`)

**Pull request merged**